contracts/cw-dao/src/contract.rs

```diff
@@ -1288,6 +1330,112 @@ mod tests {
         assert_eq!(ContractError::WrongCloseStatus {}, err.downcast().unwrap());
     }
 
+    #[test]
+    fn test_close_works_with_refund() {
+        let mut app = mock_app();
+        let voting_period = Duration::Height(2000000);
+        let threshold = Threshold::ThresholdQuorum {
+            threshold: Decimal::percent(51),
+            quorum: Decimal::percent(10),
+        };
+        let proposal_deposit_amount = Uint128::new(10);
+        let (dao_addr, _cw20_addr) = setup_test_case(
+            &mut app,
+            threshold.clone(),
+            voting_period,
+            coins(10, NATIVE_TOKEN_DENOM),
+            Some(proposal_deposit_amount),
+            Some(true),
+        );
+
+        let cw20 = Cw20Contract(_cw20_addr.clone());
+
+        let allowance = Cw20ExecuteMsg::IncreaseAllowance {
+            spender: dao_addr.clone().into(),
+            amount: proposal_deposit_amount,
+            expires: None,
+        };
+
+        let res = app.execute_contract(Addr::unchecked(OWNER), _cw20_addr.clone(), &allowance, &[]);
+        assert!(res.is_ok());
```

Comment:
Initial DAO Balance was around 10 (could change to be higher).

My impression was that the user proposing gives the DAO money so that we can then create a proposal (with which the dao can use). However if that is wrong..

Reply:
They need to send deposit funds along with the create proposal TX. CC @ben2x4.

Reply:
> My impression was that the user proposing gives the DAO money so that we can then create a proposal (with which the dao can use). However if that is wrong..

My bad, this totally works. Guess I had using `send` on Cw20... but allowance works just as well. Silly me.